apps/web/src/hooks/Trades.ts

```diff
@@ -111,18 +111,11 @@ export function useTradeExactIn(
           null
         )
       }
-      // search through trades with varying hops, find best trade out of them
-      let bestTradeSoFar: Trade<Currency, Currency, TradeType> | null = null
-      for (let i = 1; i <= MAX_HOPS; i++) {
-        const currentTrade: Trade<Currency, Currency, TradeType> | null =
-          Trade.bestTradeExactIn(allowedPairs, currencyAmountIn, currencyOut, { maxHops: i, maxNumResults: 1 })[0] ??
-          null
-        // if current trade is best yet, save it
-        if (isTradeBetter(bestTradeSoFar, currentTrade, BETTER_TRADE_LESS_HOPS_THRESHOLD)) {
```

Comment:
holding it for now. Was overlooked the trade threshold for gas.

Reply:
We can make the `bestTradeExactIn` call return multiple results and then apply the `isTradeBetter` to include the trade threshold for gas. 

Although I believe the threshold might be too low.

Reply:
I believe it's doable. It could be a new function as we don't want to have breaking change for existing functions

Reply:
I added a function to include the trade threshold.